### ISSUE TITLE
gatsby-transformer-sharp or gatsby-theme-blog doesnt seem to work well with yarn workspaces

### ISSUE THREAD

**Issue by sw-yx** [gatsbyjs/gatsby#16142]
## Description

i tried to turn a working gatsby blog using gatsby-theme-blog (verified that it works, including all the hidden dependencies inside the `content` folder) into a folder within a yarn workspace, but then the sharp plugin started throwing errors as seen below.

this may be due to my unfamiliarity with yarn workspaces but i got stuck so i figured i would file an issue. see steps

i dont have much idea of the root cause of this, my guess is that it is some sort of hardcoded path resolution assumption that fails when i work within a yarn workspaces context. couldnt trace it and i'm about to pivot to another strategy so i figured i would file this issue and leave it to experts to investigate

### Steps to reproduce

- clone https://github.com/sw-yx/browse-gatsby-themes/tree/reproWorkspaceBug
- install deps
- `yarn workspace gatsby-theme-blog start`

### Expected result

a working gatsby blog

### Actual result

```
yarn workspace v1.17.0
yarn run v1.17.0
$ gatsby develop
lksjdlksjdlks /Users/swyx/Work/browse-gatsby-themes/gatsby-theme-blog/content/posts
success open and validate gatsby-configs - 0.048 s
success load plugins - 0.033 s
success onPreInit - 0.017 s
success initialize cache - 0.020 s
success copy gatsby files - 0.075 s
success onPreBootstrap - 0.024 s
warn The gatsby-theme-blog plugin has generated no Gatsby nodes. Do you need it?
success source and transform nodes - 0.045 s
success building schema - 0.137 s
success createPages - 0.046 s
success createPagesStatefully - 0.041 s
success onPreExtractQueries - 0.011 s
success update schema - 0.041 s

 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFixed".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:37


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFixed".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:47


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFixed".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFixed".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:56


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFixed".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFixed".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:55


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFluid".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:37


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFluid".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:47


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFluid".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFluid".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:56


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFluid".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpFluid".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:55


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpResolutions".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:43


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpResolutions".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:53


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpResolutions".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:52


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpResolutions".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:62


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpResolutions".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:52


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpResolutions".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:61


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpSizes".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:37


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpSizes".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:47


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpSizes".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpSizes".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:56


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpSizes".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown type "ImageSharpSizes".

File: ../node_modules/gatsby-transformer-sharp/src/fragments.js:5:55

success extract queries from components - 0.840 s
success write out requires - 0.087 s
success write out redirect data - 0.017 s
success onPostBootstrap - 0.016 s
⠀
info bootstrap finished - 3.277 s
⠀
success run static queries - 0.016 s
success run page queries - 0.050 s — 2/2 63.57 queries/second
 DONE  Compiled successfully in 2574ms                                                                                                      11:02:22 PM
⠀
You can now view gatsby-theme-blog in the browser.
```

### Environment

```

  System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.13.0 - ~/.nvm/versions/node/v10.13.0/bin/node
    Yarn: 1.17.0 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.142
    Firefox: 67.0.4
    Safari: 12.1.1
```

**Comment by sw-yx** [comment]
as a bonus, `yarn workspace gatsby-theme-notes start` will also show you another bug where the mdx fails to be detected.

**Comment by sw-yx** [comment]
additionally, i would also make use of your new schema customization api to set a good example.

**Comment by okbel** [comment]
same here, can reproduce

**Comment by davifantasia** [comment]
I'm dealing with this issue but I'm converting to a theme from a starter. I discovered the error while I was trying to pass options from sites through `gatsby-config.js`, the error is thrown when I set config as a function instead of an object like so:

`module.exports = themeOptions =>({...});`

**Comment by nadrane** [comment]
@sw-yx Did you ever resolve this issue? I have a very similar issue where Gatsby is not finding fragments across workspaces. I'm curious what you did to solve it

**Comment by sw-yx** [comment]
pretty much just gave up

**Comment by pramam** [comment]
On installing `gatsby-plugin-sharp`, and `gatby-transformer-sharp` (and adding them to gatsby-config.js without any options) I got a similar error ERROR #85901 which had:
Unknown type: "ImageSharpFluid"
Unknown type: "ImageSharpFixed"
Unknown type: "ImageSharpSizes"

I wasn't as yet using any ImageSharp queries in my code, so this sounded like something these plugins needed was missing. 

I was following #4 Create Pages Using GraphQL here: https://www.gatsbyjs.org/docs/why-gatsby-uses-graphql/
He installs gatsby-image as a dependency(?) after installing gatsby-plugin-sharp and gatsby-transformer-sharp, but doesn't add it to gatsby-config.js.
Once I installed gatsby-image (but did not add it to gatsby-config.js), the #85901 errors about unknown types disappeared.

I hope this helps.
